File: cinema_client_interface/src/main/java/cn/edu/aynu/rjxy/client/service/ClientService.java
package cn.edu.aynu.rjxy.client.service;


import cn.edu.aynu.rjxy.pojo.TbClient;
import com.github.pagehelper.PageInfo;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ClientService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbClient> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbClient client);


	/**
	 * 修改
	 */
	public void update(TbClient client);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbClient findOne(Long clientId);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] clientIds);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbClient client, int pageNum, int pageSize);

    boolean  sendCode(String email);
}

File: client_manage_web/src/main/webapp/js/service/clientService.js
//服务层
app.service('clientService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../client/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../client/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../client/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../client/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../client/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../client/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../client/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
});